### PR TITLE
Add cosign attachment signing and verification

### LINKED ISSUE #610: Allow sbom signing and verification during attach and download
**Description**

It would be great is `cosign attach sbom` and `cosign download sbom` allowed for signing and verification of the related sbom.

Example - 

`cosign attach sbom -key <key> sbom.json` (Signs the sbom and uploads the signature to image:<sha256>.sbom.sig)

`cosign download sbom -key <key> sbom.json` (Checks if the signature exists and is valid before downloading)

Open questions - 

1. How to disable/enable signing/verification
2. Initial signing modes to support

Implementation -

Was thinking of reusing the `SignCmd` and calling it from `attach` and similarly for `verify` (`Exec`) and `download`. (We could initially support keypair signing and only pass those options in and later add others?)

### PR DESCRIPTION
**Pull request by samj1912**
Signed-off-by: Sambhav Kothari <sambhavs.email@gmail.com>

Fixes #610

### CODE REVIEW

File: cmd/cosign/cli/sign.go [hunk 1]
```diff
@@ -166,7 +167,12 @@ EXAMPLES
 			if len(args) == 0 {
 				return flag.ErrHelp
 			}
-
+			switch *attachment {
+			case "none", "sbom":
```
Comment (dlorenc) [inline cmd/cosign/cli/sign.go:171]: What do you think of keeping a list of these near https://github.com/sigstore/cosign/blob/main/pkg/cosign/fetch.go#L57? That way we could verify against it.

**Review by dlorenc — COMMENTED** [review]
Looks great! One comment about the suffix.

File: cmd/cosign/cli/verify.go [hunk 2]
```diff
@@ -110,6 +112,13 @@ func (c *VerifyCommand) Exec(ctx context.Context, args []string) (err error) {
 		return flag.ErrHelp
 	}
 
+	switch c.Attachment {
+	case "none", "sbom":
```
Comment (samj1912) [inline cmd/cosign/cli/verify.go:116]:
```suggestion
	case "sbom", "":
```

File: cmd/cosign/cli/verify.go [hunk 3]
```diff
@@ -52,6 +53,7 @@ func applyVerifyFlags(cmd *VerifyCommand, flagset *flag.FlagSet) {
 	flagset.StringVar(&cmd.RekorURL, "rekor-url", "https://rekor.sigstore.dev", "address of rekor STL server")
 	flagset.BoolVar(&cmd.CheckClaims, "check-claims", true, "whether to check the claims found")
 	flagset.StringVar(&cmd.Output, "output", "json", "output format for the signing image information (default JSON) (json|text)")
+	flagset.StringVar(&cmd.Attachment, "attachment", "none", "related image attachment to sign (none|sbom), default none")
```
Comment (samj1912) [inline cmd/cosign/cli/verify.go:56]:
```suggestion
	flagset.StringVar(&cmd.Attachment, "attachment", "", "related image attachment to sign (none|sbom), default none")
```

File: cmd/cosign/cli/sign.go [hunk 4]
```diff
@@ -166,7 +167,12 @@ EXAMPLES
 			if len(args) == 0 {
 				return flag.ErrHelp
 			}
-
+			switch *attachment {
+			case "none", "sbom", "":
```
Comment (samj1912) [inline cmd/cosign/cli/sign.go:171]:
```suggestion
			case "sbom", "":
```

**Review by dlorenc — APPROVED** [review]

**Pull request merged**